examples_deepspeed/universal_checkpointing/tb_analysis/utils.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import os
+from uc_analysis import UniversalCheckpointingAnalysis
+
+
+def find_files_recursive(directory, file_affix):
+    """
+    Recursively searches for files with a specific affix in a directory.
+
+    Args:
+        directory (str): The path to the directory to search.
+        file_affix (str): The desired file affix
+
+    Returns:
+        list: A list of paths to matching files.
+    """
+    matching_paths = []
+    for root, _, files in os.walk(directory):
```

Comment:
The function name indicates that it is recursive. Do we mean to drop the directory list and not search sub directories?

Reply:
Updated the function name and description to remove the notion of recursion of the function itself and instead just reference that `os.walk()` is used to search for files.